### ISSUE TITLE
Bug on resolve dependencies

### ISSUE THREAD

**Issue by workfel** [nestjs/nest#940]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

Hello, i just saw a bug on resolve dependencies of service on version `5.0.0-beta.6`.
I have a module who provide a service, this service inject another service. So i declare my module like this.
```ts

@Module({
  providers: [SubscriptionsService],
  exports: [SubscriptionsService],
})
export class SubscriptionsModule {
  static forRoot(port: number = 3001): DynamicModule {
    const providers = createSubscriptionProviders(port);
    return {
      module: SubscriptionsModule,
      providers: [
        ...providers,
      ],
      exports: [
        ...providers,
      ],
    };
  }
}
```
And my service 
```ts

@Injectable()
export class SubscriptionsService implements OnModuleDestroy {
  private subscriptionServer: SubscriptionServer;

  constructor(@Inject(SUBSCRIPTION_SERVER) private readonly ws) {
  }

  createSubscriptionServer(schema: any, options: ServerOptions = {}, socketOptions: WebSocket.ServerOptions = {}) {
    this.subscriptionServer = new SubscriptionServer(
      {
        execute,
        subscribe,
        schema,
        ...options,
      },
      {
        server: this.ws,
        path: '/subscriptions',
        ...socketOptions,
      },
    );
  }

  onModuleDestroy() {
    this.ws.close();
  }
}
```

When i'm doing this, i have this error
`[Nest] 51302   - 8/2/2018, 3:52:13 PM   [ExceptionHandler] Nest can't resolve dependencies of the SubscriptionsService (?). Please make sure that the argument at index [0] is available in the current context.
`

But if modify the declaration of module by removing the declaration on providers on Decorator and using it on forRoot function, it's working.

```ts

@Module({
})
export class SubscriptionsModule {
  static forRoot(port: number = 3001): DynamicModule {
    const providers = createSubscriptionProviders(port);
    return {
      module: SubscriptionsModule,
      providers: [
        ...providers,
        SubscriptionsService,
      ],
      exports: [
        ...providers,
        SubscriptionsService,
      ],
    };
  }
}
```

## Environment

<pre><code>
Nest version: 5.0.0-beta.6

 
For Tooling issues:
- Node version: 8.11.3
- Platform:  Mac
</code></pre>

**Comment by kamilmysliwiec** [comment]
As far as I remember, it was a bug fixed together with #671. It's a part of the 5.2.0 release, which will be published soon.

**Comment by 200688523** [comment]
Hello, may I ask if your problem is solved? I also have Nest can't resolve dependencies of the LOGGER_PRO (?). Please make sure that the argument at index [0] is available in the AppModule context.
